main.go

```diff
@@ -23,6 +24,9 @@ func main() {
 	serverConf := server.Config{
 		HTTPListenAddr: conf.Server.Address,
 		HTTPListenPort: conf.Server.Port,
+		HTTPMiddleware: []middleware.Interface{
+			gateway.NewAuthentication(&conf),
```

Comment:
```suggestion
```
The /metrics endpoint needs to be unauthenticated. This should work without authentication:
```
curl -v http://127.0.0.1:8080/metrics
```